### ISSUE TITLE
MarkDuplicatesSpark produces an empty bam file when run on a stand alone spark

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/gatk#1448]
This was reported by @geetduggal in #1444.  

If you run MarkDuplicatesSpark using a local standalone spark cluster it can run to completion, but produces an empty bam file that has only a header.

```
./gatk-launch MarkDuplicatesSpark -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -O testoutput.bam -- --sparkRunner SPARK --sparkMaster spark://wm1b0-8ab:7077
```

produces an empty bam

**Comment by droazen** [comment]
Duplicate of https://github.com/broadinstitute/gatk/issues/1444? :)

**Comment by lbergelson** [comment]
#1444 is several related issues.  This is more specific.  1444 depends on this.

**Comment by lbergelson** [comment]
So I've tracked this down to being a problem with line 193 in `ReadSparkSink`.  

```
       finalOut.saveAsNewAPIHadoopFile(outputFile, SAMRecord.class, SAMRecordWritable.class, SparkHeaderlessBAMOutputFormat.class);
```

saveAsNewAPIHadoopFile is not writing any part files when `outputFile` (a `String`) is a relative path.  i.e. `test.bam`.  

If you change it to a fully specified path  i.e. `/Users/louisb/Workspace/gatk/testoutput.bam` then it works as expected.

I discovered this by running the driver in the debugger which is possible using 

```
--driver-java-options -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005
``` 

as an option to spark-submit (or gatk-launch)

The quick fix is to have input paths be converted into absolute paths always.  Also to add a check in `mergeBam()` to make sure that there's at least 1 part file specified.

The longer fix is to 
1. understand why this is happening only in certain spark configurations
2. figure out how to unit test for this
